src/_cffi_src/openssl/dh.py

```diff
@@ -9,7 +9,20 @@
 """
 
 TYPES = """
-typedef ... DH;
+typedef struct dh_st {
+    /* Prime number (shared) */
+    BIGNUM *p;
+    /* Generator of Z_p (shared) */
+    BIGNUM *g;
+    /* Private DH value x */
+    BIGNUM *priv_key;
+    /* Public DH value g^x */
+    BIGNUM *pub_key;
+    /* X9.42/RFC 2631 */
+    BIGNUM *q;
+    BIGNUM *j;
+    ...;
+} DH;
```

Comment:
In OpenSSL 1.1.0 the DH_get0_pqg, DH_set0_pqg, DH_get0_key, DH_set0_key (https://www.openssl.org/docs/manmaster/crypto/DH_get0_pqg.html) functions were added. But I think that porting this struct will be more practical then using them.